### PR TITLE
Windows 11 Notepad: announce status bar if shown

### LINKED ISSUE #13688: Windows 11 Notepad: status bar is not announced
Hi,

While a bug report would be a better category, this is marked as a feature request due to the background information provided below.

### Background
As part of modernizing inbox apps (programs shipping with Windows by default), Notepad was updated on Windows 11 to adopt the new UI changes. In a recent blog post, Microsoft highlighted changes introduced in Notepad.

The new Notepad was introduced to everyone in February 2022. This version (11.2112) did introduce redesigned UI elements and use of rich edit control for document content. However, it had issues, notably screen readers such as NVDA repeating content from the line above when there was a line of text followed by a blank line, then end of the document. This, together with text repetition problem when Enter key is pressed, was resolved in Notepad 11.2203 released in late April.

One issue is not resolved: status bar reading. In Windows 10 Notepad, one can perform status bar command (NVDA+End in desktop layout, NVDA+Shift+End in laptop layout) and status bar text will be announced if enabled from View menu. This is not the case in Windows 11.

### Is your feature request related to a problem? Please describe.
Due to UI redesign, NVDA cannot obtain status bar information in Windows 11 Notepad. Further, status bar is located on different UIA tree location (fifth child of the foreground window in 11.2112, eighth child in 11.2203).

### Describe the solution you'd like
Implement status bar retrieval method in the new Notepad app module. Due to the fact that newer Notepad includes bug fixes and is generally available to everyone, tree traversal route will folow that of 11.2203 (that is, minimum supported version will be 11.2203 released in April 2022).

### Describe alternatives you've considered
Two alternatives:

1. Leave things as is. This is because Notepad app module is the example app module used in dev docs. Introducing a new Notepad module will require looking for a new simple app to provide examples for developers getting started with app modules. Alternatively, continue using Notepad with a note that for folks using Windows 11, a different app should be used to test app module basics.
2. Improve api.getStatusBar function to take situations like Windows 11 Notepad into account when fetching status bars.

### Additional context
A working example is included in Windows App Essentials add-on which will be the basis of a follow-up pull request.

Thanks.

### PR DESCRIPTION
**Pull request by josephsl**
### Link to issue number:
Fixes #13688 

### Summary of the issue:
With the redesigned Windows 11 Notepad, users cannot obtain statsu bar information.

### Description of how this pull request fixes the issue:
Introduce a new Windows 11 Notepad app module (notepad.exe). With the new UI design, status bar cannot be obtained as in prior Notepad releases. Thankfully it is possible to traverse the UIA tree to obtain the status bar object from a specific child element position (this is because child elements have the same UIA class name). Also, status bar is shown if focused on the document window (rich edit) and enabled from view menu.

### Testing strategy:
Manual testing:

1. With Windows 11 installed, install Notepad 11.2203 or later.
2. Open Notepad.
3. Select View/status bar.
4. Type text.
5. Perform status bar command (NVDA+End in desktop layout/NVDA+Shift+End in laptop layout).

### Known issues with pull request:
None, although status bar position may need to be changed if Microsoft updates Notepad interface in future releases.

### Change log entries:
Bug fixes:

NVDA will announce status bar content in Windows 11 Notepad. (#13386 

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

File: source/appModules/notepad.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		# This does not affect earlier versions.
+		notepadVersion = int(self.productVersion.split(".")[0])
+		if notepadVersion < 11:
+			raise NotImplementedError()
+		# And no, status bar is shown when editing documents.
+		# Thankfully, of all the UIA objects encountered, document window has a unique window class name.
+		if api.getFocusObject().windowClassName != "RichEditD2DPT":
+			raise NotImplementedError()
+		# Look for a specific child as some children report the same UIA properties such as class name.
+		statusBar = api.getForegroundObject().children[7].firstChild
```
Comment (seanbudd) [inline source/appModules/notepad.py:27]: Could you add a brief note on how you determined that this is the index? Is there any MS documentation for this?

```suggestion
		# Look for a specific child as some children report the same UIA properties such as class name.
		notepadStatusBarIndex = 7
		statusBar = api.getForegroundObject().children[notepadStatusBarIndex].firstChild
```

File: source/appModules/notepad.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+While this app module also covers older Notepad releases,
+this module provides workarounds for Windows 11 Notepad."""
+
+import appModuleHandler
+import api
+
+
+class AppModule(appModuleHandler.AppModule):
+
+	def _get_statusBar(self):
```
Comment (seanbudd) [inline source/appModules/notepad.py:16]: Could you add a docstring here - here's what comes from super()

```suggestion
	def _get_statusBar(self):
		"""
		Retrieve the status bar object of the application.
		If C{NotImplementedError} is raised, L{api.getStatusBar} will resort to
		a lookup by position.
		"""
```

Comment (josephsl) [reply source/appModules/notepad.py:16]: Done - I'll also add a note that status bar child position must be checked whenever Microsoft publishes Notepad updates to stable Windows 11 build users (Notepad 11 is updated through Microsoft Store as opposed to being tied to a Windows release).

Comment (seanbudd) [reply source/appModules/notepad.py:16]: Can this aspect be included? It's helpful to understand what raising `NotImplementedError` will do in this function.

```
		If C{NotImplementedError} is raised, L{api.getStatusBar} will resort to
		a lookup by position.
```

**Comment by josephsl** [comment]
Hi, done.

**Review by seanbudd — APPROVED** [review]

**Comment by josephsl** [comment]
Hi, will add a note about that (a combination of manual object navigation and Python Console was used to determine the index; the older version of Notepad 11 placed the status bar somewhere different in the UIA tree, so I used the same approach then). Thanks.

**Pull request merged**